plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -443,10 +445,22 @@ private static long getSnapshotIdFromVersion(ConnectorSession session, Table tab
 
     private static long getTargetSnapshotIdFromVersion(Table table, ConnectorTableVersion version, io.trino.spi.type.Type versionType)
     {
-        if (versionType != BIGINT) {
+        long snapshotId;
+        if (versionType == BIGINT) {
+            snapshotId = (long) version.getVersion();
+        }
+        else if (versionType instanceof VarcharType) {
```

Comment:
Do you have a plan to support `CharType` in follow-up?

Reply:
I did not originally, but I don't see why we wouldn't want to support it. Should I create a tracking issue for follow up or should I just address here?

Reply:
file an issue + add TODO comment here linking to it.

Reply:
https://github.com/trinodb/trino/issues/19125